### PR TITLE
Upkeep

### PR DESCRIPTION
**Pull request by dunkenwg**
In test-zzz-package.R, line 49 `zrxp <- ts_translate_stations(zrxp)` gives the following warning:
```
Warning (test-zzz-package.R:49:3): package
the following stations are unrecognised: 'ALH_TurbineFlow', 'Arrow_Nakusp_Elevation', 'Arrow_NonPowerFlow', 'Arrow_TailwaterElevation', 'Arrow_TotalReleases', 'Duncan_ForebayElevation', 'Fauquier_Elevation', 'Mica_ForebayElevation', 'Mica_Kinbasket_Elevation', 'Mica_NonPowerFlow', 'Mica_TurbineFlow', 'Queensbay_Elevation', 'Revelstoke_NonPowerFlow', 'Revelstoke_TailwaterElevation' and 'Revelstoke_TurbineFlow'
```
Should this warning be suppressed?

We're getting more of these warnings:
```
Warning (test-zzz-package.R:59:3): package
SQL statements must be issued with dbExecute() or dbSendStatement() instead of dbGetQuery() or dbSendQuery().
```
even though we're using `dbSendStatement()`.

### CODE REVIEW

**Comment by dunkenwg** [comment]
Would you like me to remove the R-CMD-checks that are failing? It looks like our code works properly on the updated version of R but fails on older versions because of differences in function call requirements.

**Comment by joethorley** [comment]
I've added Evan to review these changes.

**Review by evanamiesgalonski — CHANGES_REQUESTED** [review]
Yes the warning for stations can be suppressed as its calling `ts_translate_stations` on data that only has a subset of the usual stations.

The SQL warning is coming from calls to a sister package `tsbdr::ts_add_data` so you can ignore for now.  I’ll file an issue in the other repo and figure out whats going on when i have some time.

As for disabling CMD checks for older OS releases im not up to date on what our current standard is for that but as far as i know this is mostly an internal package so it might be reasonable to avoid spending time on accommodating other OSs. @joethorley thoughts?

However if we want to fix actions instead of disabling:
```Error in `as.POSIXct.numeric(span$Start, tz = "Etc/GMT+8")`: 'origin' must be supplied```

can be fixed by adding the argument `origin = "1970-01-01”` to each call of `as.POSIXct.numeric`

```Error in `check_data(data, values = list(TIME = "", BRD_FLOWS_AVG = 1,  BRD_QSPILL_AVG = 1, BRX_FLOW_AVG = 1), x_name = file)`: `names(D:/a/tscbh/tscbh/check/tscbh.Rcheck/tscbh/exdata/brd2.csv)` must include 'TIME’.```

 is likely a problem with how `utils::read.csv` behaves on older windows, it might be fixed by using a different package to read the csv like `readr::read_csv`, which would then need to get added to the imports field.

**Comment by joethorley** [comment]
Yes for this package I think we could go just back one version if @aylapear agrees.

File: .github/workflows/R-CMD-check.yaml [hunk 1]
```diff
@@ -19,17 +25,24 @@ jobs:
...
-          - {os: ubuntu-latest,   r: 'oldrel-1'}
+          # use 4.0 or 4.1 to check with rtools40's older compiler
+          - {os: windows-latest, r: 'oldrel-4'}
+
+          - {os: ubuntu-latest,  r: 'devel', http-user-agent: 'release'}
+          - {os: ubuntu-latest,  r: 'release'}
+          - {os: ubuntu-latest,  r: 'oldrel-1'}
+          - {os: ubuntu-latest,  r: 'oldrel-2'}
+          - {os: ubuntu-latest,  r: 'oldrel-3'}
+          - {os: ubuntu-latest,  r: 'oldrel-4'}
```
Comment (aylapear) [inline .github/workflows/R-CMD-check.yaml:38]: This looks like the check full template, I'd use the check standard or check release template.

**Review by aylapear — CHANGES_REQUESTED** [review]

**Review by joethorley — APPROVED** [review]

**Review by aylapear — APPROVED** [review]
Looks good to me and the new guidance you are writing up on Depends and actions hopefully provided clarity on this item.

**Pull request merged**